src/XrdTls/XrdTlsTempCA.cc

```diff
@@ -217,6 +234,26 @@ bool CRLSet::atLeastOneValidCRLFound() const {
     return m_atLeastOneValidCRLFound;
 }
 
+bool CRLSet::processCRLWithCriticalExt() {
+  // Don't open the output file if not necessary
+  if(!m_crls_critical_extension.empty()) {
+    file_smart_ptr outputfp(fdopen(dup(m_output_fd), "w"), &fclose);
```

Comment:
Please use XrdSysFD_Dup() instead of dup() to avoid FD leakage (see line 104 in the original file). This should also be corrected in line 180 that somehow missed using XrdSysFD_Dup() in CRLSet::processFile(). I know it's not technically part of this patch but it should be corrected either via this patch or with an addition pull request (technically cleaner) so that we don't forget.

Reply:
Hi @abh3,

Thanks for your very pertinent comment, I'll correct that and ship it within the same PR, but in a different commit :)
Then we are good to go!

Cheers,
Cedric

Reply:
About your question, I don't know why, but I believe it can be changed...

Reply:
If you want me to change this, I can do it :) But probably in a different PR then ;)

Reply:
I actually did it as it did not take a lot of time. I will submit a different PR once this one has been approved and merged to master